### PR TITLE
feat(house): let owners edit guest lists with aleta sio

### PR DESCRIPTION
**Pull request by caioavidal**
### Description
Adds the first Phase 3 house spell: owners and subowners can cast leta sio inside a house to open the guest list editor, save changes, and evict players no longer invited.

### Key Changes
- House window protocol (0x97 / 0x8A) with edit-session tracking and save handler
- Minimal Lua House API plus setEditHouse / sendHouseWindow for spell scripts
- invite_guests.lua spell (leta sio) and Phase 3 step doc for this slice
- Guest/subowner list saves kick uninvited occupants; door access when no per-door list exists

### Types of Changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation fix (typos, incorrect content, missing content, etc.)
- [ ] Code refactoring
- [ ] Merge Down

### Test Case
- dotnet build src/Standalone --configuration Release
- dotnet test tests/NeoServer.Server.Tests --filter FullyQualifiedName~PlayerEditHouseAccessListCommandTest --configuration Release

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/Extensions/NeoServer.Scripts.LuaJIT/NeoServer.Scripts.LuaJIT.csproj [hunk 1]
```diff
@@ -37,6 +37,7 @@
         <ProjectReference Include="..\..\ApplicationServer\NeoServer.Server.Events\NeoServer.Server.Events.csproj"/>
         <ProjectReference Include="..\..\ApplicationServer\NeoServer.Server.Helpers\NeoServer.Server.Helpers.csproj"/>
         <ProjectReference Include="..\..\Core\NeoServer.Domain\NeoServer.Domain.csproj"/>
+        <ProjectReference Include="..\..\Database\NeoServer.Data.InMemory.DataStores\NeoServer.Data.InMemory.DataStores.csproj"/>
```
Comment (caioavidal) [inline src/Extensions/NeoServer.Scripts.LuaJIT/NeoServer.Scripts.LuaJIT.csproj]: **Architecture:** This new project reference is only needed because `IHouseEditWindowStore` lives in the data-store implementation project instead of the domain contracts. If the interface moves to `NeoServer.Domain.Common.Contracts.DataStores` (alongside `IHouseStore`), the Lua extension keeps depending on contracts only — no direct reference to the in-memory store project.

File: src/Core/NeoServer.Domain/Common/Contracts/DataStores/IHouseEditWindowStore.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using NeoServer.Domain.Common.Contracts.Creatures;
+
+namespace NeoServer.Data.InMemory.DataStores;
+
+/// <summary>
+///     Tracks active house access-list edit windows per player (UI session state).
+/// </summary>
+public interface IHouseEditWindowStore
```
Comment (caioavidal) [inline src/Core/NeoServer.Domain/Common/Contracts/DataStores/IHouseEditWindowStore.cs:9]: **Architecture:** Store interfaces in this codebase live in `NeoServer.Domain.Common.Contracts.DataStores` (see `IHouseStore`, `IItemTypeStore`, etc.), but `IHouseEditWindowStore` was defined in the in-memory implementation project. That forces consumers (`PlayerFunctions`, `PlayerHouseWindowHandler`) to reference `NeoServer.Data.InMemory.DataStores` directly, and the Lua extension had to gain a new project reference for it. Please move the interface (and `HouseEditWindowSession`) to the domain contracts project so consumers depend on the contract, not the concrete store.

File: tests/NeoServer.Server.Tests/Commands/PlayerEditHouseAccessListCommandTest.cs [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        command.Execute(owner, houseId: 10, HouseListId.GuestList, text: string.Empty);
+
+        // Assert
+        houseRepository.Verify(x => x.SaveAccessList(10, HouseListId.GuestList, string.Empty), Times.Once);
+        eviction.Verify(
+            x => x.TeleportToExit(guest, It.Is<Location>(l => l.X == 50 && l.Y == 50 && l.Z == 7)),
+            Times.Once);
+    }
+
+    [Fact]
```
Comment (caioavidal) [inline tests/NeoServer.Server.Tests/Commands/PlayerEditHouseAccessListCommandTest.cs:72]: **Testing:** Missing `[Trait("Category", ...)]` on 3 of the 4 tests in this file (this one, `Execute_PlayerCannotEdit_DoesNotSave` at line 99, and `HouseEditWindowStore_MismatchedWindowId_ReturnsFalse` at line 122). Every `[Fact]` should carry a category trait (e.g., `HappyPath`, `Validation`, `ErrorCondition`). Also note the mocking policy: `IHouseEviction` is a domain service, and per AGENTS.md only `I*Repository` and infrastructure interfaces should be mocked — prefer a real `HouseEvictionService` (with a stubbed movement service) or document why the mock is needed.

File: src/ApplicationServer/NeoServer.Server.Commands/Player/House/PlayerEditHouseAccessListCommand.cs [hunk 4]
```diff
@@ -59,5 +63,50 @@ public void Execute(IPlayer player, uint houseId, uint listId, string text)
...
         houseRepository.SaveAccessList(houseId, listId, text);
+
+        KickUninvitedPlayers(house, listId);
+    }
+
+    /// <summary>
+    ///     After guest/subowner list changes, kick anyone no longer invited.
+    ///     Door-list edits do not kick.
+    /// </summary>
+    private void KickUninvitedPlayers(Domain.Houses.House house, uint listId)
```
Comment (caioavidal) [inline src/ApplicationServer/NeoServer.Server.Commands/Player/House/PlayerEditHouseAccessListCommand.cs]: **Design (DDD):** The eviction workflow — scanning house tiles, deciding which occupants are no longer invited, then teleporting them — is business logic that lives in the command. Per AGENTS.md, business logic belongs in domain services (`*Service`), not in handlers/commands. Suggest moving the occupant scan + eviction into the house domain (e.g., an `IHouseEviction.KickUninvited(house, ...)` method or a `House.GetUninvitedOccupants()`), keeping the command as a thin orchestrator. Also, `Domain.Houses.House` is redundant here since `using NeoServer.Domain.Houses;` is already imported.

**Review by caioavidal — COMMENTED** [review]
## Code Review — `feat(house): let owners edit guest lists with aleta sio`

Reviewed against AGENTS.md conventions. **Build ✅** (`dotnet build src/Standalone -c Release`, 0 errors) and **tests ✅** (PlayerEditHouseAccessListCommandTest + HouseTileItemSerializerTest: 8/8; HouseDoorAccessTests: 11/11).

### What's good

- Well-scoped Phase 3 slice: protocol (0x97/0x8A), edit-session store, save path, minimal Lua surface, and a working spell — all coherently wired (InputHandlerMap, LuaStartup, IoC modules).
- Thread affinity respected: the save executes through `IDispatcher`, and the command re-validates `CanEditAccessList` on every save, so the window/session is not a privilege escalation vector.
- `PlayerHouseWindowHandler` validates the session windowTextId before applying edits (mirrors `PlayerWriteEventHandler`).
- Door-access behavior change is documented in code comments and covered by updated domain tests (empty door list still denies; no list → invited guests allowed).
- Docs: `house-phase3-steps.md` is a nice tracking artifact; `enum_functions.lua` globals match `HouseListId` (0x100/0x101).

### Findings (inline comments posted)

1. **Architecture — store interface placement** (`IHouseEditWindowStore.cs`, `NeoServer.Scripts.LuaJIT.csproj`): `IHouseEditWindowStore` lives in the in-memory implementation project instead of `NeoServer.Domain.Common.Contracts.DataStores` where every other store interface (`IHouseStore`, etc.) lives. This forces the Lua extension to add a direct project reference to `NeoServer.Data.InMemory.DataStores`. Move the interface (+ `HouseEditWindowSession`) to domain contracts.
2. **DDD — eviction logic in command** (`PlayerEditHouseAccessListCommand.cs:74`): the "who is no longer invited → evict" workflow is business logic that belongs in the domain (e.g., `IHouseEviction`/house service), keeping the command a thin orchestrator. Also `Domain.Houses.House` is redundant with the existing `using`.
3. **Testing — missing traits** (`PlayerEditHouseAccessListCommandTest.cs`): 3 of 4 tests lack `[Trait("Category", ...)]`; also `Mock<IHouseEviction>` (a domain service) slightly exceeds the repository/infrastructure-only mocking policy, and names (`Execute_...`) don't follow `{Actor}_does_{something}_when_{condition}`.
4. **Threading (minor, informational)** (`PlayerHouseWindowHandler.cs`): `HouseEditWindowStore` is a plain `Dictionary`-backed store touched from the network thread (`TryGet`/`Clear`) and the game thread (dispatcher event). Matches the pre-existing `ItemTextWindowStore` pattern, so not blocking — just worth documenting the assumption.

### Notes (no action strictly required)

- **Behavior change**: `House.CanUseDoor` + `LuaActionScriptService.CanUseHouseDoor` now let any *invited* player use house doors when no per-door list exists (and doors lacking a DoorId). Intended per the PR, but it's a guest-enters-through-any-door semantic — confirm this matches the expected house behavior before merge.
- **Map change**: `data/world/small.otbm` is modified (binary) — commit message says "update small test map" but the PR description doesn't mention it. Worth one line in the description (which house/door tiles were added).
- **Coverage**: no test yet for `PlayerHouseWindowHandler` / packet flow (functional-testing skill could cover handler → command via the test container), and none for the Lua bindings — optional follow-up.
- **Commit hygiene**: two of three commits are conventional (`feat(house):`, `refactor(house):`); "Add unit test trait" could be `test(house):`.

Overall: **approve with suggestions** — the feature is sound, tested, and well-documented; the findings are convention/layering improvements rather than blockers.

File: src/Core/NeoServer.Domain/Common/Contracts/DataStores/IHouseEditWindowStore.cs (hunk 2 above)
Comment (caioavidal) [reply src/Core/NeoServer.Domain/Common/Contracts/DataStores/IHouseEditWindowStore.cs:9]: Resolved in 6179425f9 — moved to `NeoServer.Domain.Common.Contracts.DataStores` following the `IHouseStore` pattern; the LuaJIT project reference was dropped and all consumers now depend on the domain contract. A deeper refactor of this session-state design is planned separately.

File: src/Extensions/NeoServer.Scripts.LuaJIT/NeoServer.Scripts.LuaJIT.csproj (hunk 1 above)
Comment (caioavidal) [reply src/Extensions/NeoServer.Scripts.LuaJIT/NeoServer.Scripts.LuaJIT.csproj]: Resolved in 6179425f9 — the reference was removed; the Lua extension now depends on the domain contract only.

File: tests/NeoServer.Server.Tests/Commands/PlayerEditHouseAccessListCommandTest.cs (hunk 3 above)
Comment (caioavidal) [reply tests/NeoServer.Server.Tests/Commands/PlayerEditHouseAccessListCommandTest.cs:72]: Fixed in 65fbfb3: all four tests now carry a Category trait, and the mocked IHouseEviction was replaced with the real HouseEvictionService backed by a recording fake movement service (FakeCreatureMovementService) so assertions verify the actual eviction flow.

Comment (caioavidal) [reply tests/NeoServer.Server.Tests/Commands/PlayerEditHouseAccessListCommandTest.cs:72]: Follow-up in 06f558a: no fakes or service mocks remain. The tests now assemble real implementations from the helper suite — real Map + Player (MapTestDataBuilder/PlayerTestDataBuilder), real CreatureMovementService (with real CylinderOperation, CreatureMovementValidation and StaticToDynamicTileService built from ItemFactoryTestBuilder + in-memory stores), real HouseEvictionService, and the real in-memory HouseStore. HouseTestDataBuilder gained a realTiles parameter so houses link actual map tiles. Only IHouseRepository stays mocked, per the repository-only policy.
